paper/paper.md

```diff
@@ -0,0 +1,103 @@
+  - counterfactual explanation
+  - recourse
+authors:
+  - name: Hangzhi Guo
+    orcid: 0009-0000-6277-9003
+    corresponding: true # (This is how to denote the corresponding author)
+    affiliation: 1 # (Multiple affiliations must be quoted)
+  - name: Xinchang Xiong
+    affiliation: 2
+  - name: Wenbo Zhang
+    affiliation: 1
+  - name: Amulya Yadav
+    orcid: 0009-0005-4638-9140
+    affiliation: 1
+affiliations:
+ - name: Penn State University, USA
+   index: 1
+ - name: Duke University, USA
+   index: 2
+date: 10 December 2023
+bibliography: paper.bib
+
+# Optional fields if submitting to a AAS journal too, see this blog post:
+# https://blog.joss.theoj.org/2018/12/a-new-collaboration-with-aas-publishing
+# aas-doi: 10.3847/xxxxx <- update this with the DOI from AAS once you know it.
+# aas-journal: Astrophysical Journal <- The name of the AAS journal.
+---
+
+# Summary
+
+Counterfactual explanation[^1] techniques provide contrast cases to individuals adversely affected by ML preictions.
+For instance, recourse methods can provide suggestions for loan applicants who have been rejected by a bank's ML algorithm, or give practical advice to teachers handling students at risk of dropping out.
+Numerous recourse explanation methods have been recently proposed.
+Yet, current research practice focuses on medium-sized datasets (typically around ~50k data points).
+This limitation impedes the progress in algorithmic recourse and raises concerns about the scalability of existing approaches. 
+
+[^1]: Counterfactual explanation [@wachter2017counterfactual] and algorithmic recourse [@ustun2019actionable] share close connections [@verma2020counterfactual;@stepin2021survey], which leads us to use these terms interchangeably
+
+
+To address this challenge, we propose `ReLax`, a JAX-based benchmarking library, designed for efficient and scalable recourse explanations. `ReLax` supports various recourse methods and datasets, demonstrating performance improvements of at least two orders of magnitude over current libraries.
+Notably, ReLax can benchmark real-world datasets up to 10 million data points, a 200-fold increase over existing norms, without imposing prohibitive computational costs.
+
+
+
+# Statement of need
+
+
+Recourse and counterfactual explanation methods concentrate on the generation of new instances that lead to contrastive predicted outcomes [@verma2020counterfactual;@karimi2020survey;@stepin2021survey]. Given their ability to provide actionable recourse, these explanations are often favored by human end-users [@binns2018s;@miller2019explanation;@Bhatt20explainable].
+
+
+Despite progress made in counterfactual explanation research [@wachter2017counterfactual;@mothilal2020explaining;@ustun2019actionable;@upadhyay2021towards;@vo2023feature;@guo2021counternet;@guo2023rocoursenet], current research practices often restrict the evaluation of recourse explanation methods on medium-sized datasets (with under 50k data points). 
```

Comment:
Start with many of these methods are slow

Reply:
I am going to leave this paragraph as-is because it is more about the slowness of implementation in those libraries. It is not because these methods are slow. It is mostly because they are implemented inefficiently.